ibm_vpd_app.cpp

```diff
@@ -918,6 +923,26 @@ static void populateDbus(T& vpdMap, nlohmann::json& js, const string& filePath)
         const auto& objectPath = item["inventoryPath"];
         sdbusplus::message::object_path object(objectPath);
 
+        vector<string> ccinList;
+        if (item.find("ccin") != item.end())
+        {
+            for (const auto& cc : item["ccin"])
+            {
+                string ccin = cc;
```

Comment:
is this copy really necessary?  is cc not already a std::string?

Reply:
Yes, it is necessary. The build failed when I used cc. There seems to be some difference as I tried changing from 'auto' to 'std::string' as well. Ended-up with having an extra variable :(

Reply:
I made this change for upstream- https://gerrit.openbmc-project.xyz/c/openbmc/openpower-vpd-parser/+/48853/5/ibm_vpd_app.cpp#1013
and I am able to build it without any extra variable.

Reply:
@Alpana07  the build fails when you try to transform the string to uppercase. Yes, it does work to push it into the vector.

Reply:
ok - I think the json data structure itself must be const.  So at least one copy is necessary.